Content.Shared/Damage/Components/DamageableComponent.cs

```diff
@@ -329,37 +308,153 @@ public void Heal()
                 foreach (var type in types)
                 {
                     var damageAmount = Math.Min(damagePerType, damageLeft);
-                    ChangeDamage(type, damageAmount, true);
+                    ChangeDamage(type, damageAmount, ignoreDamageResistances, source, extraParams);
                     damageLeft -= damageAmount;
                 }
             }
 
             return true;
         }
 
-        public bool SetDamage(DamageTypePrototype type, int newValue, IEntity? source = null, DamageChangeParams? extraParams = null)
+
+        // TODO QUESTION Please dont run away, its a very interesting wall of text.
```

Comment:
Yeah, I'd probably replace ChangeDamage(DamageGroupPrototype) with this function.

Reply:
brain hurts ill read this and give my thoughts later

Reply:
I've replaced it with my alternate version. Also fixed a potential divide-by-zero error.